### ISSUE TITLE
Can

### ISSUE THREAD

**Issue by jaymegordo** [executablebooks/MyST-Parser#402]
```
/Users/Jayme/OneDrive/MDS/591/Orbis-Investments-Capstone/CONTRIBUTING.md:4: WARNING: 'myst' reference target not found: README.md
```

**To Reproduce**

**Expected behavior**

A link to your documentation repository.

If relevant, a minimal example of the input text should be supplied,
together with a screen-shot of the output document and/or command-line output, e.g.

```markdown
some text...
```

```console
$ jupyter-book build mybook
ERROR ...
```

**Environment**

- Python Version [e.g. 3.7.1]:
- Package versions or output of `jupyter-book --version`:
- Operating System:

**Additional context**

**Comment by chrisjsewell** [comment]
heys @jaymegordo could you provide a minimal working example to reproduce this warning, i.e. it seems like you have something like this

```
docs/source/conf.py
    extensions = ["myst_parser"]
docs/source/index.md
    content?
CONTRIBUTING.md
    # Heading

    [README](README.md)
README.md
    # Heading
```

**Comment by vineetsharma883** [comment]
I have following folder structure:-
```
automation_p/
    |_____worfklow.md
    |_____Conditional.sh
docs/
    |____source
    |     |_____conf.py
    |     |_____index.rst
    |_____build
```

Following is problematic line in `workflow.md` - 
```
...
...
to check the condition ref [Conditional.sh](./Conditional.sh)
...
...
```

On running `make html`, I am getting following error - `WARNING: 'myst' reference target not found: ./Conditional.sh`

**Comment by choldgraf** [comment]
I believe that the issue here is that `.sh` files are not supported by Sphinx, and so it cannot find it. If you want to link to a raw file, use ``{download}`./Conditional.sh <Conditional.sh>` ``

**Comment by arwedus** [comment]
I guess that the need of the original author is to suppress these warnings, not to fix the issue itself.
I have the same request.

I include some files in the Sphinx documentation which are targeted at people who navigate the source code in an IDE or in a github or Bitbucket GUI. I include links to certain files which should not be copied to the documentation output, e.g. a header file or source file, to support easy navigation in the code.

The warnings we get are along the lines of:

```
/home/user/workspace/package/README.md:13: WARNING: 'myst' reference target not found: ./CMakeLists.txt
/home/user/workspace/package/README.md:13: WARNING: 'myst' reference target not found: ./src/main.cpp
```

Wanted behavior: Just remove the link and output the text (if available) or the link.

**Comment by arwedus** [comment]
@chrisjsewell : Are you still actively working on MyST-parser? I have not seen much activity of you since the last release (1.5.2 End of August). This topic is still important to us. Who will pick up the development?

**Comment by chrisjsewell** [comment]
> Are you still actively working on MyST-parser?

Yes, but I'm afraid this is just not a priority fix for myself, since it is not it is not keeping with sphinx itself:
you cannot reference arbitrary files with sphinx `doc`/`any` roles, only parsed files.
As noted by @choldgraf, you can use roles like `download` to do this

I think you can probably suppress all reference warnings, I don't know off-hand what the warning type is (it would be nice is sphinx itself provided these like myst does, but unfortunately they don't https://github.com/sphinx-doc/sphinx/issues/8845)

**Comment by chrisjsewell** [comment]
Naturally, we are happy to receive PRs with proposals to add such features

**Comment by john-hen** [comment]
I think the problem here is that a warning is raised in the first place, not to suppress it. And it's because the syntax is ambiguous. We can do `` `Conditional.sh <./Conditional.sh>`_ `` in reStructuredText and would expect this to be the same as `[Conditional.sh](./Conditional.sh)` in Markdown. No Sphinx roles involved whatsoever. Of course, if the link target is a source file, such as `ReadMe.md`, then MyST should try to resolve that reference, so it acts like the `any` role. But this is not a problem in reST/Sphinx itself.

**Comment by choldgraf** [comment]
An issue here is that Sphinx only copies over the files that are explicitly included as a part of the document model, so if you did `[mylink](some-file.txt)`, then when your documentation HTML is built, there's no guarantee that `some-file.txt` will be there to link to in your docs. The `{download}` role will cause a *copy* to happen for the target file, so that it's included with the assets in your outputs.

So it's a bit more complex than just telling Sphinx not to raise a warning if the file is not a source file.

**Comment by john-hen** [comment]
I agree that the use case is somewhat esoteric. But I just happened to answer a [question] on SO today where the user was asking for a way around the `download` role. And there is a way. We can use [`html_extra_path`] to have Sphinx copy over files, at least to the root folder. I guess I must have used something like that myself at some point. Though not sure about the exact use case of the other commenters in this thread. It's not really clear to me either how those files get there.

[question]: https://stackoverflow.com/q/70158364
[`html_extra_path`]: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_extra_path

**Comment by arwedus** [comment]
> I think the problem here is that a warning is raised in the first place, not to suppress it. And it's because the syntax is ambiguous. We can do `` `Conditional.sh <./Conditional.sh>`_ `` in reStructuredText and would expect this to be the same as `[Conditional.sh](./Conditional.sh)` in Markdown. No Sphinx roles involved whatsoever. Of course, if the link target is a source file, such as `ReadMe.md`, then MyST should try to resolve that reference, so it acts like the `any` role. But this is not a problem in reST/Sphinx itself.

Exactly! In rST there's a difference between a file you just link to (i.e. ``my source file <../src/main.cpp>`_`) and documents you refer to (i.e. `:doc:'index'`). rST won't check plain links and just don't care. The use case for this is to refer to some files in the same repo, and the links are navigatable e.g. via an rST viewer in VS Code or on github.

In Markdown, I just have links, whether it's `[other doc](path/to/other_doc.md)` or `[my source file](../src/main.cpp)`. apparently, MyST tries to transform all markdown links to `:doc:` references.

The issue pops up here since we want to include already existing Markdown documents with our developer documentation in sphinx - and they are full of links to source files etc.

@chrisjsewell : Anyhow, I'm happy and relieved to read that MyST Parser is still under active development / maintainance and you would be willing to review PRs. Very good 👍🏼 keep up the good work.

**Comment by mara004** [comment]
Which warning type do I need to suppress to hide `'myst' reference target not found` and `Unknown target name` warnings?

**Comment by ravindk89** [comment]
@mara004 what worked for me is:


```
suppress_warnings = [ 'ref.myst']
```

**Comment by mara004** [comment]
Thanks!

**Comment by linusjf** [comment]
> [@mara004](https://github.com/mara004) what worked for me is:
> 
> ```
> suppress_warnings = [ 'ref.myst']
> ```

Doesn't work for me

**Comment by mara004** [comment]
@linusjf This thread is relatively old. IIRC, something has changed since then.
Does `suppress_warnings = ['myst.xref_missing']` work?
See also https://github.com/pypdfium2-team/pypdfium2/commit/f145e5e82b71cbf8d6ee8280f5e4283605d0e99f

**Comment by linusjf** [comment]
> [@linusjf](https://github.com/linusjf) This thread is relatively old. IIRC, something has changed since then. Does `suppress_warnings = ['myst.xref_missing']` work? See also [pypdfium2-team/pypdfium2@f145e5e](https://github.com/pypdfium2-team/pypdfium2/commit/f145e5e82b71cbf8d6ee8280f5e4283605d0e99f)

Yes,that does. I overlooked adding that after discovering the new code.
